R/utils.R

```diff
@@ -49,11 +49,16 @@ generate_random_id <- function(prefix, id_length = 20) {
 #'
 #' This throws warning if there are parameters not supported by semantic.
 #'
-#' @param args list with extra arguments
+#' @param args list or vector with extra arguments
 check_extra_arguments <- function(args) {
```

Comment:
I don't understand the purpose of this function. It says it will throw a warning if parameters are not supported. But it looks like anything passed as `args` here will be either stopped ("wrong input type") or warning when length(args) > 0. Could you explain it to me? :)

Reply:
Yes, you can imagine the situation that user migrates from shiny to semantic and uses some button:

```r
b1 <- button(a=1, b=2)
library(shiny.semantic)
b2 <- button(a=1, b=2) # this is semantic button
# throws warning that b is not supported argument in semantic, but still creates b2
```

There are arguments that are hard to directly map to `shiny.semantic` behaviour, so we just warn the user that the bahaviour might be different than expected, but we don't want to interrupt execution of the app. That was the idea.

Reply:
1. Maybe the name of the function is confusing, but the documentation comment says "This throws warning if there are parameters not supported by semantic.". Where is it checked in the function code? It looks like any vector (length > 1) will return a warning. This is more like a function for constructing warning message, not a function that checks arguments, am I right?

2. I still need to ask - what should happen when I run:

```
check_extra_arguments(list(a=1))
```

Why there should be no warning?

Maybe I am missing something here :)